src/lib/support/PersistentData.h

```diff
@@ -49,13 +49,23 @@ struct PersistentData
         TLV::TLVWriter writer;
         writer.Init(mBuffer, sizeof(mBuffer));
 
-        ReturnErrorOnFailure(Serialize(writer));
-
-        // Save serialized data
-        return storage->SyncSetKeyValue(key.KeyName(), mBuffer, static_cast<uint16_t>(writer.GetLengthWritten()));
+        CHIP_ERROR err = Serialize(writer);
```

Comment:
Why these changes?  This seems logically equivalent to what we had before....

Reply:
No functional change here. I have found myself making this change every time I need to debug these objects. Reverted.